### PR TITLE
vencord: 1.3.4 -> 1.4.5

### PR DESCRIPTION
**Pull request by Scrumplex**
## Description of changes

https://github.com/Vendicated/Vencord/compare/v1.3.4...v1.4.5

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by NotAShelf — APPROVED** [review]

File: pkgs/misc/vencord/default.nix [hunk 1]
```diff
@@ -50,6 +50,11 @@ buildNpmPackage rec {
     ./disable-updater-ui.patch
   ];
 
+  postPatch = ''
+    # Prevent build scripts from including this file in builds
+    rm src/plugins/_core/settings.tsx.orig
```
Comment (FlafyDev) [inline pkgs/misc/vencord/default.nix:55]: Do you know why it's being created in the first place?
It seems there's also `./scripts/build/common.mjs.orig`
And there are no `./src/utils/updater.ts.orig` and `./src/components/VencordSettings/UpdaterTab.tsx.orig`.

Comment (Scrumplex) [reply pkgs/misc/vencord/default.nix:55]: It's possible that the patch doesn't apply cleanly (line numbers might have changed) which is why patch might leave the original file there as a backup.

I could try updating the patches to see if that fixes it

Comment (Scrumplex) [reply pkgs/misc/vencord/default.nix]: That was indeed the case!

**Review by FlafyDev — APPROVED** [review]
Works in `webcord-vencord`

File: pkgs/misc/vencord/replace-git.patch [hunk 2]
```diff
@@ -1,17 +1,17 @@
 diff --git a/scripts/build/common.mjs b/scripts/build/common.mjs
-index 7ff599a..85b3bfa 100644
+index b63ea61..10e4ee1 100644
 --- a/scripts/build/common.mjs
 +++ b/scripts/build/common.mjs
-@@ -24,7 +24,7 @@ import { promisify } from "util";
- 
+@@ -33,7 +33,7 @@ export const VERSION = PackageJSON.version;
+ export const BUILD_TIMESTAMP = Date.now();
```
Comment (delroth) [inline pkgs/misc/vencord/replace-git.patch:6]: Correct me if I'm wrong, but if this runs at build time, it doesn't seem [reproducible](https://reproducible-builds.org/)?

Could you use [ `SOURCE_DATE_EPOCH`](https://reproducible-builds.org/docs/source-date-epoch/) from the environment instead? I think it's what you want. (And that sounds like a patch you could send upstream too - `SOURCE_DATE_EPOCH` with `Date.now()` as fallback when not set.)

**Review by delroth — CHANGES_REQUESTED** [review]

**Review by delroth — APPROVED** [review]
Changing my review to not block merging - I feel like the change I'm asking for is important, but AFAIK nixpkgs doesn't have a well defined policy *requiring* reproducibility (though it's a goal that many people work towards, see https://reproducible.nixos.org/)

**Comment by Scrumplex** [comment]
I have opened a PR to upstream our patches.

See https://github.com/Vendicated/Vencord/pull/1677

**Pull request merged**